pkg/trace/test/testutil/stats.go

```diff
@@ -22,7 +23,7 @@ func BucketWithSpans(spans []*stats.WeightedSpan) pb.ClientStatsBucket {
 	for _, s := range spans {
 		// override version to ensure all buckets will have the same payload key.
 		s.Meta["version"] = ""
-		srb.HandleSpan(s, defaultEnv, defaultHostname, defaultContainerID)
+		srb.HandleSpan(s, "", defaultEnv, defaultHostname, defaultTracerHostname, defaultContainerID)
```

Comment:
The way this looks is an indication that there should not be multiple hostname args here. Only one, final hostname. The logic outside of these functions should compute it, since they know the answer and the variables. Related to https://github.com/DataDog/datadog-agent/pull/9757/files#r743507394